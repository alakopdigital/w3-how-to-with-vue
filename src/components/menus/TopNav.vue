<template lang="html">
  <nav class="top-nav">
    <slot></slot>
  </nav>
</template>

<script>
export default {
  name: 'top-nav'
}
</script>

<style lang="scss">
.top-nav {
  width: 100%;
  background-color: #333;
  overflow: hidden;

  a {
    float: left;
    display: block;
    color: #f2f2f2;
    text-align: center;
    padding: 14px 16px;
    text-decoration: none;
    font-size: 17px;
    cursor: pointer;

    &:hover {
      background-color: #ddd;
      color: black;
    }

    &.is-active {
      background-color: #4CAF50;
      color: white;
    }
  }
}
</style>
