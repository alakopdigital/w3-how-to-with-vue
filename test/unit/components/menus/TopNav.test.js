import {mount, renderer} from '#/unit/utils'
import TopNav from '@/components/menus/TopNav.vue'

describe('TopNav.vue', () => {
  it('render default', () => {
    const wrap = mount(TopNav)

    expect(wrap.hasClass('top-nav')).toBe(true)

    renderer.renderToString(wrap.vm, (err, str) => {
      if (err) console.error(err)
      expect(str).toMatchSnapshot()
    })
  })

  it('slots#default', () => {
    const wrap = mount(TopNav, {
      slots: {
        default: `<a>Test</a>`
      }
    })

    expect(wrap.hasClass('top-nav')).toBe(true)
    expect(wrap.find('a').text()).toEqual('Test')

    renderer.renderToString(wrap.vm, (err, str) => {
      if (err) console.error(err)
      expect(str).toMatchSnapshot()
    })
  })
})
