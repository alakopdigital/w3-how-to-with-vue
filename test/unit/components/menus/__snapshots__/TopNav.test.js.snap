// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopNav.vue render default 1`] = `<nav class="top-nav"></nav>`;

exports[`TopNav.vue slots#default 1`] = `<nav class="top-nav"><a>Test</a></nav>`;
